src/awkward/operations/ak_to_parquet.py

```diff
@@ -178,6 +178,62 @@ def to_parquet(
     # Dispatch
     yield (array,)
 
+    return _impl(
+        array,
+        destination,
+        list_to32,
+        string_to32,
+        bytestring_to32,
+        emptyarray_to,
+        categorical_as_dictionary,
+        extensionarray,
+        count_nulls,
+        compression,
+        compression_level,
+        row_group_size,
+        data_page_size,
+        parquet_flavor,
+        parquet_version,
+        parquet_page_version,
+        parquet_metadata_statistics,
+        parquet_dictionary_encoding,
+        parquet_byte_stream_split,
+        parquet_coerce_timestamps,
+        parquet_old_int96_timestamps,
+        parquet_compliant_nested,  # https://issues.apache.org/jira/browse/ARROW-16348
+        parquet_extra_options,
+        storage_options,
+        iter=False,
```

Comment:
`iter` is a builtin Python function name, one that you might want to use in the implementation (if it doesn't fit well into a single `for` loop). By using it as the boolean back-door, you lose access to that function. (You can get it back with `__builtins__.iter`, but that's deliberately making things difficult.)

```python
>>> def generator():
...     for _ in range(5):
...         print("making another one")
...         yield "another one"
... 
>>> it = iter(generator())
>>> next(it)
making another one
'another one'
>>> next(it)
making another one
'another one'
>>> next(it)
making another one
'another one'
>>> next(it)
making another one
'another one'
>>> next(it)
making another one
'another one'
>>> next(it)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
StopIteration
```